pkg/pulumiyaml/ast/template.go

```diff
@@ -268,6 +268,16 @@ func (d *PropertyMapDecl) parse(name string, node syntax.Node) syntax.Diagnostic
 	return diags
 }
 
+type ConfigItemsDecl struct {
+	declNode
+
+	Type *StringExpr
+}
```

Comment:
Do we also need to include values here/is that even supported?

Reply:
Ah good spotting! I think this can be arbitrarily nested, we can use `ConfigParamDecl` recursively here, updated the PR!